core/src/main/java/org/elasticsearch/indices/cluster/IndicesClusterStateService.java

```diff
@@ -267,376 +302,264 @@ protected void doRun() throws Exception {
             }
         }
 
-        for (IndexService indexService : indicesService) {
-            IndexMetaData indexMetaData = event.state().metaData().index(indexService.index());
+        // delete local indices that do neither exist in previous cluster state nor part of tombstones
+        for (AllocatedIndex<? extends Shard> indexService : indicesService) {
+            Index index = indexService.index();
```

Comment:
since we have the AllocatedIndex here can't we get the IndexMetaData from the `AllocatedIndex#getIndexSettings()`